src/GitHub.App/Services/ModelService.cs

```diff
@@ -427,39 +451,98 @@ public AccountCacheItem Owner
             public DateTimeOffset UpdatedAt { get; set; }
         }
 
+        [NullGuard(ValidationFlags.None)]
         public class PullRequestCacheItem : CacheItem
         {
             public static PullRequestCacheItem Create(PullRequest pr)
             {
-                return new PullRequestCacheItem(pr);
+                return new PullRequestCacheItem(pr, new PullRequestFile[0]);
+            }
+
+            public static PullRequestCacheItem Create(PullRequest pr, IList<PullRequestFile> files)
+            {
+                return new PullRequestCacheItem(pr, files);
             }
 
             public PullRequestCacheItem() {}
+
             public PullRequestCacheItem(PullRequest pr)
+                : this(pr, new PullRequestFile[0])
+            {
+            }
+
+            public PullRequestCacheItem(PullRequest pr, IList<PullRequestFile> files)
             {
                 Title = pr.Title;
                 Number = pr.Number;
+                Base = new GitReferenceModel { Label = pr.Base.Label, Ref = pr.Base.Ref, RepositoryCloneUrl = pr.Base.Repository.CloneUrl };
+                Head = new GitReferenceModel { Label = pr.Head.Label, Ref = pr.Head.Ref, RepositoryCloneUrl = pr.Head.Repository.CloneUrl };
```

Comment:
It's possible for the `pr.Head.Repository` property to be null causing a `NullReferenceException` here. For example, check out this PR: https://github.com/github/Scientist.net/pull/20 After it was merged, the user deleted the repository.